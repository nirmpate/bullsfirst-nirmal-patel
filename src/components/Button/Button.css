.button {
    border-radius: 4px;
    padding: 5px 16px 5px 16px;
    min-height: 30px;
    min-width: 77px;
    cursor: pointer;
    font-family: var(--primary-font);
    font-style: var(--primary-font-style);
    font-weight: normal;
    font-size: 16px;
    line-height: 26px;
    border: 1px solid transparent;
}

.button--outline {
    background-color: transparent;
}

.button--filled.button--primary {
    color: var(--white);
    background-color: var(--primary-800);
}

.button--filled.button--secondary {
    color: var(--neutral-900);
    background-color: var(--secondary-400);
}

.button--outline.button--primary {
    color: var(--neutral-900);
    border: 1px solid var(--neutral-900);
}

.button--outline.button--secondary {
    color: var(--secondary-900);
    border: 1px solid var(--secondary-900);
}