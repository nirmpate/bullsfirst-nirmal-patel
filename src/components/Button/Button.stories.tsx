// Button.stories.js

import { Button, ButtonType } from './Button';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
      onClick: {action: 'clicked' },
  }
} as Meta


const Template: Story<ButtonType> = (args) => <Button {...args}> Button</Button>;

export const ButtonComponentStory = Template.bind({});
ButtonComponentStory.args = {
    type: 'button',
    variant: 'filled',
    color: 'primary'
}
