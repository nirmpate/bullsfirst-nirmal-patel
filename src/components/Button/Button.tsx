import './Button.css';

export interface ButtonType {
    variant: 'filled' | 'outline';
    color: 'primary' | 'secondary';
    
    /** Link or Button Element */
    type: 'link' | 'button',
    children?: string;

    /** Href URL */
    linkHref?: string;

    /** Simple Click Handler */
    onClick?: (event: any) => {}
}

export const Button = (props: ButtonType) => {
    const { variant, color, type, children, linkHref, onClick } = props;

    let classList = ['button'];
    
    variant === 'filled' ? classList.push('button--filled') : classList.push('button--outline');
    color === 'primary' ? classList.push('button--primary') : classList.push('button--secondary');

    if (type === 'link') {
        return (<a className={ classList.join(" ") } href={ linkHref }> { children }</a>);
    } else {
        return (<button className={ classList.join(" ") } onClick={onClick}> { children }</button>);
    }
}