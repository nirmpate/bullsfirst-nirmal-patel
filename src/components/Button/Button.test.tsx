import { render, screen } from '@testing-library/react';
import { Button, ButtonType } from './Button';

const defaultArgs: ButtonType = {
  type: 'button',
  variant: 'filled',
  color: 'primary',
};

test('renders correctly', () => {
  const { asFragment } = render(<Button {...defaultArgs}>Test Button </Button>);
  expect(asFragment()).toMatchSnapshot();
});

test('filled outline outline variant', () => {
  const outlineArgs: ButtonType = {
    type: 'button',
    variant: 'outline',
    color: 'primary',
  };

  render(<Button {...outlineArgs}>Test Button </Button>);
  const button = screen.getByRole('button');

  expect(button.className).toEqual('button button--outline button--primary');
});

test('outline secondary outline variant', () => {
  const outlineArgs: ButtonType = {
    type: 'button',
    variant: 'outline',
    color: 'secondary',
  };

  render(<Button {...outlineArgs}>Test Button </Button>);
  const button = screen.getByRole('button');

  expect(button.className).toEqual('button button--outline button--secondary');
});

test('filled primary filled primary variant', () => {
  const outlineArgs: ButtonType = {
    type: 'button',
    variant: 'filled',
    color: 'primary',
  };

  render(<Button {...outlineArgs}>Test Button </Button>);
  const button = screen.getByRole('button');

  expect(button.className).toEqual('button button--filled button--primary');
});

test('filled secondary outline variant', () => {
  const outlineArgs: ButtonType = {
    type: 'button',
    variant: 'filled',
    color: 'secondary',
  };

  render(<Button {...outlineArgs}>Test Button </Button>);
  const button = screen.getByRole('button');

  expect(button.className).toEqual('button button--filled button--secondary');
});
