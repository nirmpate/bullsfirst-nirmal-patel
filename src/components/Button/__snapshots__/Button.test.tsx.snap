// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button button--filled button--primary"
  >
     Test Button 
  </button>
</DocumentFragment>
`;
